tests/bless/basic_math.c

```diff
@@ -0,0 +1,21 @@
+// file: basic_math.3cfc46df15d2d47-cgu.0.c
+#include <stdint.h>
+
+/* Some helper macros for the generated code */
+
+/** cast from unsigned to signed
+  * example: `__rust_utos(uint32_t, int32_t, x, INT32_MAX)`
+  */
+#define __rust_utos(u, s, v, m) \
+    ((v) <= (m) ? ((s)v) : ((s)((u)(v) - (u)(m) - 1)))
+
+int32_t main();
+int64_t foo(uint8_t _0, uint16_t _1, uint32_t _2);
+
+int32_t main() { return 0; }
```

Comment:
Why is `int32_t` generated rather than `int`? On some (uncommon) platforms, `int` is 16 bits.

Reply:
This originates from Rust's `i32`.

 In Rust, there is no exact equivalent to C's `int` type (`core::ffi::c_int` is as an alias for either `i32` or `i16`, depending on the platform, and it is resolved before code generation).

Reply:
Ah makes sense, thanks for the clarification.